syncmonkey/src/main/java/com/chesapeaketechnology/syncmonkey/SharingActivity.java

```
@@ -0,0 +1,341 @@
+                if (cut != -1)
+                {
+                    result = result.substring(cut + 1);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * @param uris The list of URIs representing the files.
+     * @return All the file names from the list of file URIs as one concatenated String with the names separated by commas.
+     */
+    private String getFileNamesAsString(List<Uri> uris)
+    {
+        final StringBuilder fileNames = new StringBuilder();
+        uris.forEach(uri -> fileNames.append(getFileName(uri)).append(", "));
+
+        return fileNames.toString();
+    }
+
+    /**
+     * Takes the current shared file(s) and copies them to the private shared directory.
+     */
+    private void copySharedFilesToSyncMonkeyDirectory()
+    {
+        try
+        {
+            if (sharedImageUri != null)
+            {
+                // A single image was shared
+
+                final EditText fileNameEditText = findViewById(R.id.fileNameValue);
+                final String fileName = fileNameEditText.getText().toString();
+                if (fileName.isEmpty())
+                {
+                    Toast.makeText(getApplicationContext(), R.string.sharing_error_invalid_file_name, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                copySharedFile(sharedImageUri, fileName);
+
+                Toast.makeText(getApplicationContext(), R.string.sharing_success_file, Toast.LENGTH_SHORT).show();
+            } else if (sharedImageUris != null)
+            {
+                // Multiple images were shared
+
+                sharedImageUris.forEach(imageUri -> {
+                    try
+                    {
+                        copySharedFile(imageUri, null);
+                    } catch (Exception e)
+                    {
+                        Log.e(LOG_TAG, "Unable to copy the file in the Sync Monkey Sharing Activity due to an exception: ", e);
```

By catching the error here, it will fail silently and the user won't get a toast if one of the files causes an exception to be thrown.

---

Good point.  I will update it.